packages/core/src/connectivity/scp-cf/destination/destination-from-service.ts

```diff
@@ -497,56 +504,38 @@ class DestinationFromServiceRetriever {
   }
 
   private isProviderNeeded(
-    resultFromSubscriber: DestinationSearchResult
+    resultFromSubscriber: DestinationSearchResult | undefined
   ): boolean {
-    if (this.options.selectionStrategy === alwaysSubscriber) {
-      return false;
-    }
-    if (
-      this.options.selectionStrategy === subscriberFirst &&
-      resultFromSubscriber
-    ) {
-      return false;
-    }
-
-    return true;
+    return (
+      this.options.selectionStrategy !== alwaysSubscriber &&
```

Comment:
For me this here is not an improvement. I find it hard to process.

Reply:
I kind of agree and kind of disagree. Will revert ;)